src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/TranscriptionServiceImpl.java

```diff
@@ -250,11 +253,23 @@ private TranscriptionWorkflowEntity saveTranscriptionWorkflow(UserAccountEntity
         transcriptionWorkflow.setTranscriptionStatus(transcriptionStatus);
         transcriptionWorkflow.setWorkflowActor(userAccount);
         transcriptionWorkflow.setWorkflowTimestamp(OffsetDateTime.now(UTC));
-        transcriptionWorkflow.setWorkflowComment(workflowComment);
-        transcriptionWorkflow.setCreatedBy(userAccount);
-        transcriptionWorkflow.setLastModifiedBy(userAccount);
 
-        return transcriptionWorkflowRepository.saveAndFlush(transcriptionWorkflow);
+        TranscriptionWorkflowEntity savedTranscriptionWorkFlow = transcriptionWorkflowRepository.saveAndFlush(
+            transcriptionWorkflow);
+
+        if (!StringUtils.isBlank(workflowComment)) {
+            createAndSaveComment(userAccount, workflowComment, savedTranscriptionWorkFlow);
+        }
+        return savedTranscriptionWorkFlow;
+    }
+
+    private void createAndSaveComment(UserAccountEntity userAccount, String workflowComment, TranscriptionWorkflowEntity savedTranscriptionWorkFlow) {
+        TranscriptionCommentEntity transcriptionCommentEntity = new TranscriptionCommentEntity();
+        transcriptionCommentEntity.setComment(workflowComment);
+        transcriptionCommentEntity.setTranscriptionWorkflowId(savedTranscriptionWorkFlow.getId());
```

Comment:
Think you need to link the `tra_id` from transcription_comment also - if you could do that please?setTranscription

Reply:
ACCEPT